app/Helpers/Docker.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace App\Helpers;
+
+use Symfony\Component\Yaml\Yaml;
+use App\Exceptions\ProcessFailed;
+use LaravelZero\Framework\Commands\Command;
+use Symfony\Component\Yaml\Exception\ParseException;
+
+class Docker
+{
+
+    /** @var Helpers $helpers */
+    public $helpers;
+
+    public function __construct(Helpers $helpers)
+    {
+        $this->helpers = $helpers;
+    }
+
+    public function tagImages(Command $command, string $service = null, string $sourceTag, array $newTags): bool
```

Comment:
I didn't do this as this means I have to call `tagImages($command, null)`

Without the default, you have to meet the type hint requirements @kubatek94

Reply:
@spamoom the problem here is that default arguments must follow non default arguments (if any), otherwise if we have:

`tagImages(Command $command, string $service = null, string $sourceTag, array $newTags): bool`

when we call it like `tagImages($this)` - what values do $sourceTag and $newTags get?

It only makes sense if we have:

`tagImages(Command $command, string $sourceTag, array $newTags, string $service = null): bool`

I just don't understand why PHP doesn't fail, as the first version should be a fatal error IMO.

Reply:
Ahh with you now!

Once we get the docker version ready, we can just use PHP 8 + named arguments!

Reply:
After all of this, I looked at where it was being used and it makes sense now xD